smolyay/basis.py

```diff
@@ -403,29 +482,393 @@ def _derivative(self, x):
         float
             Value of the derivative of Trigonometric polynomial.
 
+        n : int, optional
+            order of derivative. Default is 1.
+
         Raises
         ------
         ValueError
             If input is outside the domain `[0, 2\pi]`
         """
         x = numpy.asarray(x)
-        return self.frequency * 1j * numpy.exp(x * self.frequency * 1j)
+        return numpy.exp(x * self.frequency * 1j) * (self.frequency * 1j) ** n
+
+
+class BasisFunctionSet(collections.abc.Sequence):
+    """Set of basis functions and sample points."""
+
+    @abc.abstractmethod
+    def __init__(self):
+        pass
```

Comment:
`__init__` should not be abstract. I would have this `__init__` method accept a list but default to None, then either assign the list or an empty list to `self.basis_functions`.